include/alpaka/rand/Philox/PhiloxBaseCudaPlain.hpp

```diff
@@ -0,0 +1,106 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+
+#include <alpaka/rand/Philox/mulhilo.hpp>
+
+#include <utility>
+
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace engine
+        {
+            /** Philox backend using plain CUDA uintN types for the storage of Key and Counter
+             *
+             * @tparam TParams Philox algorithm parameters \sa PhiloxParams
+             * @tparam TImpl engine type implementation (CRTP)
+             */
+            template<typename TParams, typename TImpl>
+            class PhiloxBaseCudaPlain
+            {
+                static_assert(TParams::counterSize == 4, "GPU Philox implemented only for counters of width == 4");
+
+            public:
+                using Counter = uint4;
+                using Key = uint2;
+
+                static constexpr uint64_t WEYL_64_0 = 0x9E3779B97F4A7C15; /* golden ratio */
```

Comment:
Maybe add a comment where this numbers are taken from.

Reply:
fixed in next iteration